File: 98-validate-binary-search-tree/validate-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isValidBST(TreeNode root) {
        return helper(root, null, null);

    }
    boolean helper(TreeNode root,Integer low,Integer high){

        if(root == null){
            return true;
        }
        if((low != null && low >= root.val) || (high != null && high <= root.val)){
            return false;
        }
        return helper(root.left, low, root.val) && helper(root.right, root.val, high);

    }
}

File: 98-validate-binary-search-tree/validate-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        return self.helper(root, None, None)
    def helper(self, node, low, high):
        if not node:
            return True
        if (low != None and low >= node.val) or (high != None and high <= node.val):
            return False

        return self.helper(node.left, low, node.val) and self.helper(node.right, node.val, high)


        